src/os/unix.rs

```diff
@@ -1,17 +1,27 @@
 #![cfg(any(target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd"))]
 
 use std::sync::Arc;
+use std::ptr;
 use libc;
 use Window;
 use platform::Window as LinuxWindow;
+use platform::{UnixBackend, UNIX_BACKEND};
 use WindowBuilder;
 use api::x11::XConnection;
+use api::x11::ffi::XVisualInfo;
 
 use wayland_client::protocol::wl_display::WlDisplay;
 use wayland_client::protocol::wl_surface::WlSurface;
 
 pub use api::x11;
 
+pub fn get_x11_xconnection() -> Option<Arc<XConnection>> {
```

Comment:
I don't like the fact that the `XConnection` is exposed, but the fix would be complicated so please add a `TODO: do not expose XConnection` here.